packages/tree-widget/src/components/TreeUiItemsProvider.tsx

```diff
@@ -0,0 +1,127 @@
+  public readonly id = "TreeUiitemsProvider";
+  public static i18n: I18N;
+
+  private _imodel: IModelConnection;
+  private _activeView?: Viewport;
+  private _modelsTreeProps?: {};
+  private _categoriesTreeProps?: {};
+  private _spatialTreeProps?: {};
+  private _enablePreloading?: boolean;
+  private _enableElementsClassGrouping?: boolean;
+  private _allViewports?: boolean;
+  private _additionalTrees?: SelectableContentDefinition[];
+  private _modelsTreeReplacement?: () => React.ReactNode;
+  private _categoriesTreeReplacement?: () => React.ReactNode;
+  private _spatialTreeReplacement?: () => React.ReactNode;
+
+  constructor(props: TreeWidgetControlOptions) {
+    this._imodel = props.iModelConnection;
+    this._activeView = props.activeView;
+    this._modelsTreeProps = props.additionalProps?.modelsTree;
+    this._categoriesTreeProps = props.additionalProps?.categoriesTree;
+    this._spatialTreeProps = props.additionalProps?.spatialTree;
+    this._enablePreloading = props.enablePreloading;
+    this._enableElementsClassGrouping = props.enableElementsClassGrouping;
+    this._allViewports = props.allViewports;
+    this._additionalTrees = props.additionalTrees;
+    this._modelsTreeReplacement = props.treeReplacements?.modelsTree;
+    this._categoriesTreeReplacement = props.treeReplacements?.categoriesTree;
+    this._spatialTreeReplacement = props.treeReplacements?.spatialTree;
+  }
+
+  public provideWidgets(
+    _stageId: string,
+    _stageUsage: string,
+    location: StagePanelLocation,
+    _section: StagePanelSection | undefined,
+  ): ReadonlyArray<AbstractWidgetProps> {
+    const widgets: AbstractWidgetProps[] = [];
+
+    if (location === StagePanelLocation.Right) {
+
+      const modelsTreeComponent = (
+        <ModelsTreeComponent
+          iModel={this._imodel}
+          allViewports={this._allViewports}
+          activeView={this._activeView}
+          enablePreloading={this._enablePreloading}
+          enableElementsClassGrouping={this._enableElementsClassGrouping}
+          {...this._modelsTreeProps}
+        />
+      );
+
+      const categoriesTreeComponent = (
+        <CategoriesTreeComponent
+          iModel={this._imodel}
+          allViewports={this._allViewports}
+          activeView={this._activeView}
+          enablePreloading={this._enablePreloading}
+          {...this._categoriesTreeProps}
+        />
+      );
+
+      const spatialContainmentComponent = (
+        <SpatialContainmentTree
+          iModel={this._imodel}
+          enablePreloading={this._enablePreloading}
+          enableElementsClassGrouping={
+            this._enableElementsClassGrouping
+              ? ClassGroupingOption.Yes
+              : ClassGroupingOption.No
+          }
+          {...this._spatialTreeProps}
+        />
+      );
+
+      const trees: SelectableContentDefinition[] = [
+        {
+          label: UiFramework.translate("visibilityWidget.modeltree"),
```

Comment:
I guess using `UiFramework.translate` is fine, but id rather we use `IModelApp.i18n.translate`, or use `TreeUiItemsProvider.i18n.translate`

Reply:
Actually, should use TreeWidget.i18n